supervision/metrics/detection.py

```diff
@@ -0,0 +1,420 @@
+        ...     [ 1.0, 1.0, 2.0, 2.0 ],
+        ...     ]), confidence=array([ 1.0 ]), class_id=array([2]))
+        ... ]
+        >>> predictions = [
+        ...     Detections(
+        ...         xyxy=array([
+        ...     [ 0.0, 0.0, 3.0, 3.0 ],
+        ...     [ 0.1, 0.1, 3.0, 3.0 ],
+        ...     [ 6.0, 1.0, 8.0, 3.0 ],
+        ...     [ 1.0, 6.0, 2.0, 7.0 ],
+        ...     ]),
+        ...     confidence=array([ 0.9, 0.9, 0.8, 0.8 ]),
+        ...     class_id=array([1, 0, 1, 1])
+        ...     ),
+        ...     Detections(
+        ...         xyxy=array([
+        ...     [ 1.0, 1.0, 2.0, 2.0 ]
+        ...     ]),
+        ...     confidence=array([ 0.8 ]),
+        ...     class_id=array([2])
+        ...     )
+        ... ]
+
+        >>> confusion_matrix = ConfusionMatrix.from_detections(
+        ...     predictions=predictions,
+        ...     targets=target,
+        ...     num_classes=3
+        ... )
+
+        >>> confusion_matrix.matrix
+        ... array([
+        ...     [0., 0., 0., 0.],
+        ...     [0., 1., 0., 1.],
+        ...     [0., 1., 1., 0.],
+        ...     [1., 1., 0., 0.]
+        ... ])
+        ```
+        """
+
+        prediction_tensors = []
+        target_tensors = []
+        for prediction, target in zip(predictions, targets):
+            prediction_tensors.append(cls.convert_detections_to_tensor(prediction))
+            target_tensors.append(cls.convert_detections_to_tensor(target))
+        return cls.from_tensors(
+            predictions=prediction_tensors,
+            targets=target_tensors,
+            classes=classes,
+            conf_threshold=conf_threshold,
+            iou_threshold=iou_threshold,
+        )
+
+    @classmethod
+    def convert_detections_to_tensor(cls, detections: Detections) -> np.ndarray:
+        arrays_to_concat = [detections.xyxy, np.expand_dims(detections.class_id, 1)]
+        if detections.confidence is not None:
+            arrays_to_concat.append(np.expand_dims(detections.confidence, 1))
+
+        return np.concatenate(
+            arrays_to_concat,
+            axis=1,
+        )
+
+    @classmethod
+    def from_tensors(
+        cls,
+        predictions: List[np.ndarray],
+        targets: List[np.ndarray],
+        classes: List[str],
+        conf_threshold: float = 0.3,
+        iou_threshold: float = 0.5,
+    ) -> ConfusionMatrix:
+        """
+        Calculate confusion matrix based on predicted and ground-truth detections.
+
+        Args:
+            predictions: detected objects. Each element of the list describes a single image and has shape = (M, 6) where M is the number of detected objects. Each row is expected to be in (x_min, y_min, x_max, y_max, class, conf) format.
+            target: ground-truth objects. Each element of the list describes a single image and has shape = (N, 5) where N is the number of ground-truth objects. Each row is expected to be in (x_min, y_min, x_max, y_max, class) format.
```

Comment:
Notice that you have a different name for the argument here. It should be `targets`.